File: Server/src/main/java/Server/core/game/content/dialogue/EadburgDialogue.java
package core.game.content.dialogue;

import core.game.node.entity.npc.NPC;
import core.plugin.Initializable;
import core.game.node.entity.player.Player;

/**
 * Represents the dialogue plugin used for the eadburg npc.
 * @author 'Vexia
 * @version 1.0
 */
@Initializable
public final class EadburgDialogue extends DialoguePlugin {

	/**
	 * Constructs a new {@code EadburgDialogue} {@code Object}.
	 */
	public EadburgDialogue() {
		/**
		 * empty.
		 */
	}

	/**
	 * Constructs a new {@code EadburgDialogue} {@code Object}.
	 * @param player the player.
	 */
	public EadburgDialogue(Player player) {
		super(player);
	}

	@Override
	public DialoguePlugin newInstance(Player player) {
		return new EadburgDialogue(player);
	}

	@Override
	public boolean open(Object... args) {
		npc = (NPC) args[0];
		interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "What's cooking, good looking?");
		stage = 0;
		return true;
	}

	@Override
	public boolean handle(int interfaceId, int buttonId) {
		switch (stage) {
		case 0:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "The stew for the servant's main meal.");
			stage = 1;
			break;
		case 1:
			interpreter.sendDialogues(player, FacialExpression.HALF_GUILTY, "Um...er...see you later.");
			stage = 2;
			break;
		case 2:
			interpreter.sendDialogues(npc, FacialExpression.HALF_GUILTY, "Bye!");
			stage = 3;
			break;
		case 3:
			end();
			break;
		}
		return true;
	}

	@Override
	public int[] getIds() {
		return new int[] { 1072 };
	}
}

File: Server/src/main/content/region/asgarnia/burthorpe/dialogue/EadburgDialogue.kt
package content.region.asgarnia.burthorpe.dialogue

import core.api.openDialogue
import core.game.dialogue.DialogueBuilder
import core.game.dialogue.DialogueBuilderFile
import core.game.dialogue.DialoguePlugin
import core.game.dialogue.FacialExpression
import core.game.node.entity.player.Player
import core.plugin.Initializable
import org.rs09.consts.NPCs

@Initializable
class EadburgDialogue(player: Player? = null) : DialoguePlugin(player) {

    override fun handle(interfaceId: Int, buttonId: Int): Boolean {
        openDialogue(player, EadburgDialogueFile(), npc)
        return true
    }

    override fun newInstance(player: Player?): DialoguePlugin {
        return EadburgDialogue(player)
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.EADBURG_1072)
    }
}

class EadburgDialogueFile : DialogueBuilderFile() {

    override fun create(b: DialogueBuilder) {
        b.defaultDialogue().playerl(
            FacialExpression.FRIENDLY,
            "What's cooking, good looking?"
        ).npcl(
            FacialExpression.FRIENDLY,
            "The stew for the servant's main meal."
        ).playerl(
            FacialExpression.HALF_WORRIED,
            "Um...er...see you later."
        ).npcl(
            FacialExpression.FRIENDLY,
            "Bye!"
        )
    }
}